### PR TITLE
Begin working on User page

### PR DESCRIPTION
**Pull request by Jneville0815**

### CODE REVIEW

File: src/resources/js/pages/Users.tsx [hunk 1]
```diff
@@ -1,5 +1,66 @@
...
+    });
+  };
+
+  const getUsers = async () => {
+    try {
+      const allUsers = await RootServerApi.getUsers();
+      setUsers(allUsers);
+      console.log(allUsers);
+    } catch (error: any) {
+      console.log(error);
```
Comment (jbraswell) [inline src/resources/js/pages/Users.tsx:33]: Look for examples of calling `RootServerApi.handleErrors`.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (jbraswell) [inline package-lock.json:1]: Probably delete this file.

File: src/resources/js/pages/Users.tsx [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+
+  useEffect(() => {
+    getUsers();
+  }, []);
+
+  return (
+    <div>
+      <Box>
+        <FormControl>
+          <InputLabel id='select-label'>User</InputLabel>
```
Comment (jbraswell) [inline src/resources/js/pages/Users.tsx:45]: Everwhere you add a string, add an english translation to `localization.ts`. Then you can import `strings` and use the translation.

File: src/resources/js/pages/Users.tsx (hunk 1 above)
Comment (GarrettWeinberg) [reply src/resources/js/pages/Users.tsx:33]: can do 
```
import {
Box,
FormControl,
...} from @mui/material
```

Comment (GarrettWeinberg) [reply src/resources/js/pages/Users.tsx:33]: Just my preference tho

File: src/resources/js/pages/Users.tsx [hunk 4]
```diff
@@ -1,5 +1,66 @@
...
 export const Users = () => {
-  return <div>{strings.usersTitle}</div>;
+  const [currentSelection, setCurrentSelection] = useState('create-new-user');
+  const [selectedUser, setSelectedUser] = useState<any>({});
+  const [users, setUsers] = useState<User[]>([]);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setCurrentSelection(event.target.value as string);
+    users.forEach((user) => {
+      if (event.target.value === user.username) {
```
Comment (jbraswell) [inline src/resources/js/pages/Users.tsx:19]: Would it be better to use `user.id` for the event target?

File: src/resources/js/pages/Users.tsx [hunk 5]
```diff
@@ -1,5 +1,84 @@
...
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { User } from 'bmlt-root-server-client';
+import { useEffect, useState } from 'react';
+
+import RootServerApi from '../RootServerApi';
 import { strings } from '../localization';
 
 export const Users = () => {
-  return <div>{strings.usersTitle}</div>;
+  const [currentSelection, setCurrentSelection] = useState('-1');
```
Comment (jbraswell) [inline src/resources/js/pages/Users.tsx:10]: Should you be passing the number `-1` instead of the string `'-1'`?

File: src/package-lock.json [hunk 6]
```diff
@@ -1,8790 +0,0 @@
-{
-    "name": "src",
```
Comment (jbraswell) [inline src/package-lock.json:2]: This file needs to be restored.

File: src/resources/js/localization.ts [hunk 7]
```diff
@@ -4,16 +4,22 @@ export const strings = new LocalizedStrings({
...
     loginTitle: 'Login',
+    emailTitle: 'Email',
     usernameTitle: 'Username',
     passwordTitle: 'Password',
     rootServerTitle: 'Root Server',
     dashboardTitle: 'Dashboard',
     meetingsTitle: 'Meetings',
     serviceBodiesTitle: 'Service Bodies',
     usersTitle: 'Users',
+    userTitle: 'User',
```
Comment (jbraswell) [inline src/resources/js/localization.ts:15]: Try to find translations for each of these strings from the existing translation files in the root server - I sent you the github url to those files in Slack.

File: src/resources/js/pages/Users.tsx (hunk 5 above)
Comment (Jneville0815) [reply src/resources/js/pages/Users.tsx:10]: The MenuItem value wants a string, so I just convert the ID to a string

File: src/resources/js/pages/Users.tsx [hunk 8]
```diff
@@ -1,5 +1,84 @@
...
+  if (errorMessage) {
+    return <div>{errorMessage}</div>;
+  }
+
+  return (
+    <div>
+      <Box>
+        <FormControl>
+          <InputLabel id='select-label'>{strings.userTitle}</InputLabel>
+          <Select labelId='select-label' id='select' value={currentSelection} label='User' onChange={handleChange}>
```
Comment (jbraswell) [inline src/resources/js/pages/Users.tsx:52]: the label here should use a translated string

**Review by jbraswell — APPROVED** [review]

**Pull request merged**